### PR TITLE
feat: add support for nextjs private folders

### PR DESCRIPTION
**Pull request by leepowellnbs**
This PR adds support for Nexts colocation private folders https://nextjs.org/docs/app/building-your-application/routing/colocation#private-folders

### CODE REVIEW

**Review by DukeLuo — CHANGES_REQUESTED** [review]
Can you add a windows style test case in file `folder-naming-convention.windows.js`?

**Comment by leepowellnbs** [comment]
> Can you add a windows style test case in file `folder-naming-convention.windows.js`?

Done

**Review by DukeLuo — CHANGES_REQUESTED** [review]

File: tests/lib/rules/folder-naming-convention.windows.js [hunk 1]
```diff
@@ -411,6 +411,11 @@ ruleTester.run(
         filename: 'src\\app\\(marketing)\\page.ts',
         options: [{ 'src/**/': 'NEXT_JS_APP_ROUTER_CASE' }],
       },
+      {
+        code: "var foo = 'bar';",
+        filename: 'src/app/_components/page.ts',
+        options: [{ 'src/**/': 'NEXT_JS_APP_ROUTER_CASE' }],
```
[the remark this answers is not in the document]
Comment (leepowellnbs) [reply tests/lib/rules/folder-naming-convention.windows.js:417]: My bad, missed that - should be resolved now.

**Comment by leepowellnbs** [comment]
@DukeLuo Out of interest would you have any appetite to support the private folder style for files too? Next's documentation says:
```
While not a framework convention, you might also consider marking files outside private folders as "private" using the same underscore pattern.
```
So it would be good to support something like `_server-actions.ts` too. Happy to do that as a separate PR.

**Review by DukeLuo — APPROVED** [review]

**Comment by DukeLuo** [comment]
> @DukeLuo Off the top of my head, the only two I can think of for filenames in a nextjs project would be kebab case and a private folder with kabab case. Can you think of any others?

Actually, i dont know a lot about nextjs. Based on what I know about nextjs so far, I think the answer is yes.

**Comment by DukeLuo** [comment]
@leepowellnbs Thanks for your PR. This PR has been released in v2.6.2.

**Pull request merged**